cider/exec/nextgen/operators/OpNode.h

```diff
@@ -18,54 +18,86 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-#ifndef CIDER_EXEC_NEXTGEN_TRANSLATOR_DUMMY_H
-#define CIDER_EXEC_NEXTGEN_TRANSLATOR_DUMMY_H
 
-#include <memory>
+#ifndef NEXTGEN_TRANSLATOR_OPNODE_H
+#define NEXTGEN_TRANSLATOR_OPNODE_H
+
 #include <type_traits>
-#include <utility>
-#include <vector>
 
-#include "exec/nextgen/jitlib/base/JITFunction.h"
+#include "exec/nextgen/Context.h"
 #include "type/plan/Analyzer.h"
 
-namespace cider::exec::nextgen::translator {
-using namespace cider::jitlib;
+namespace cider::exec::nextgen::operators {
+class Translator;
+class OpNode;
+
+using OpNodePtr = std::shared_ptr<OpNode>;
+using ExprPtr = std::shared_ptr<Analyzer::Expr>;
+using ExprPtrVector = std::vector<ExprPtr>;
 
-class Context {
+/// \brief A OpNode is a relational operation in a plan
+///
+/// Note: Each OpNode has zero or one source
+class OpNode : protected std::enable_shared_from_this<OpNode> {
```

Comment:
> The constructors of [std::shared_ptr](https://devdocs.io/cpp/memory/shared_ptr) detect the presence of an unambiguous and accessible (ie. public inheritance is mandatory) (since C++17) enable_shared_from_this base

Reply:
Has been fixed in next PR.